src/ol/interaction/Draw.js

```diff
@@ -905,11 +912,23 @@ class Draw extends PointerInteraction {
         }
       }
       this.geometryFunction_(coordinates, geometry, projection);
+
+      if (geometry.getType() === GeometryType.POLYGON && this.sketchLine_) {
```

Comment:
Can you please explain what this `if`-block is for? It looks to me like it just writes the coordinates of the sketch line again. Why would that be necessary? Why wasn't it necessary before?

Reply:
You can see this issue in the codesandbox in #11713  Start drawing a Star, move the mouse over the undo button and click.  The fill disappears but the sketch line does not update until the mouse moves back over the viewport.

Reply:
`removeLastPoint` should be a noop, also when using the `CIRCLE` type (currently only when using the `LINE_STRING` type. If you fix that, you can remove this block and also the additional `if (coordinates.length >= 2)` blocks you added.

Reply:
It is not specific to Circle geometry.  The Star example could equally well use LineString geometry with `maxPoints: 2` and this example https://jsfiddle.net/j0vbs2fc/ has no limit on points.

The `coordinates.length >= 2` I added fixes an existing problem.  Start drawing a polygon in https://openlayers.org/en/main/examples/draw-features.html then open the console and see the errors  being logged as you click undo.

Reply:
Ok, I see. However, I'm still worried about recreating the sketch line over and over during modification. This could make the interaction really slow when modifying a polygon with many vertices. Also, what if you're modifying an inner ring (as can be tested in the modify-test example)?

I'd really like to see this solved differently somehow.

Reply:
The extra call only happens when the application calls `removeLastPoint` and the mode is LineString and geometry type is Polygon (i.e. when a geometry function is specified), so it is not going to be something which happens often.  It is based on existing code at https://github.com/openlayers/openlayers/blob/main/src/ol/interaction/Draw.js#L826 which is unaltered and is already called frequently when geometry functions are specified.  There is no obvious performance impact even with over 30 points defining a bezier curve polygon (which is likely to have thousands of vertices in its ring) and that is probably an extreme case.

Reply:
Thanks for the explanation, @mike-000. In this case, it would probably make sense to factor that code out into a separate function, and call that function from both places where it is used.

This is the last remark I have. Great work!